colossalai/zero/zero_optimizer.py

```diff
@@ -138,45 +155,14 @@ def step(self, *args, **kwargs):
             self.zero_grad()
             self._update_fp16_params()
             return
-        self._update_params_ptr()
         ret = self.optim.step(*args, **kwargs)
         self._register_states()
         self.zero_grad()
         self._update_fp16_params()
         return ret
 
-    def compute_grad_norm(self, norm_type: float = 2.0) -> float:
-        norm_type = float(norm_type)
-        if not self.chunk_manager.enable_distributed_storage:
-            return compute_grad_norm(self.module.parameters(), norm_type)
-
-        non_distributed_params = []
-        distributed_params = []
-        for p in self.module.parameters():
-            if getattr(p, '_ddp_to_ignore', False):
-                non_distributed_params.append(p)
-            else:
-                distributed_params.append(p)
-        non_distributed_norm = _compute_grad_lp(non_distributed_params, norm_type)
-        distributed_norm_tensor = torch.tensor([_compute_grad_lp(distributed_params, norm_type)],
-                                               device=get_current_device())
-        if norm_type == inf:
-            dist.all_reduce(distributed_norm_tensor,
-                            op=dist.ReduceOp.MAX,
-                            group=self.chunk_manager.process_group.dp_process_group())
-            total_norm = max(non_distributed_norm, distributed_norm_tensor.item())
-        else:
-            dist.all_reduce(distributed_norm_tensor, group=self.chunk_manager.process_group.dp_process_group())
-            total_norm = non_distributed_norm + distributed_norm_tensor.item()
-            total_norm = total_norm**(1 / norm_type)
-        return total_norm
-
     def clip_grad_norm(self, model: torch.nn.Module, max_norm: float, norm_type: float = 2.0):
-        if self.optim_state == OptimState.SCALED:
-            self._unscale_grads()
-        total_norm = self.compute_grad_norm(norm_type)
-        _clip_grad_norm(self.module.parameters(), max_norm, total_norm)
-        return total_norm
+        raise NotImplementedError
```

Comment:
Is it difficult to implement grad clipping?

Reply:
I wonder whether we should calculate the norm of the paramters in GPU.

Reply:
You said calculating grad norm on CPU is slow, right?

Reply:
right.

Reply:
So why not calculating grad norm on CUDA?

Reply:
ok. let's adopt this feature in the next pr.